File: FleetTrack/src/FleetTrack.Domain/Entities/Vehicle.cs
using FleetTrack.Domain.Enums;

namespace FleetTrack.Domain.Entities;

public class Vehicle : BaseEntity
{
    public string RegistrationNumber { get; set; } = string.Empty;
    public string Brand { get; set; } = string.Empty;
    public string Model { get; set; } = string.Empty;
    public int Year { get; set; }
    public VehicleType Type { get; set; }
    public VehicleStatus Status { get; set; }
    public FuelType FuelType { get; set; }
    public double FuelCapacity { get; set; }
    public double CurrentFuelLevel { get; set; }
    public int Mileage { get; set; }
    public DateTime? LastMaintenanceDate { get; set; }
    public DateTime? NextMaintenanceDate { get; set; }

    // Navigation properties
    public Guid? CurrentDriverId { get; set; }
    public Driver? CurrentDriver { get; set; }
    public ICollection<Mission> Missions { get; set; } = new List<Mission>();
    public ICollection<GpsPosition> GpsPositions { get; set; } = new List<GpsPosition>();
    public ICollection<Alert> Alerts { get; set; } = new List<Alert>();
    public ICollection<Maintenance> MaintenanceRecords { get; set; } = new List<Maintenance>();
}

File: fleettrack-frontend/src/types/vehicle.ts
export enum VehicleType {
  Car = 0,
  Truck = 1,
  Van = 2,
  Motorcycle = 3,
  Bus = 4,
  Trailer = 5,
  Other = 6
}

export enum VehicleStatus {
  Available = 0,
  InUse = 1,
  Maintenance = 2,
  OutOfService = 3
}

export enum FuelType {
  Gasoline = 0,
  Diesel = 1,
  Electric = 2,
  Hybrid = 3,
  LPG = 4,
  CNG = 5,
  Hydrogen = 6
}

export interface VehicleDto {
  id: string;
  registrationNumber: string;
  brand: string;
  model: string;
  year: number;
  type: VehicleType;
  status: VehicleStatus;
  fuelType: FuelType;
  fuelCapacity: number;
  currentFuelLevel: number;
  mileage: number;
  lastMaintenanceDate?: string;
  nextMaintenanceDate?: string;
  currentDriverId?: string;
  currentDriverName?: string;
}

export interface CreateVehicleRequest {
  registrationNumber: string;
  brand: string;
  model: string;
  year: number;
  type: VehicleType;
  fuelType: FuelType;
  fuelCapacity: number;
  currentFuelLevel: number;
  mileage: number;
}

export interface UpdateVehicleRequest {
  brand: string;
  model: string;
  status: VehicleStatus;
  currentFuelLevel: number;
  mileage: number;
  nextMaintenanceDate?: string;
}
